bin/_os.sh

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+
+set -eu
+
+os() {
+  os=$(uname -s)
+  arch=""
+  case $os in
+    CYGWIN* | MINGW64*)
+      os=windows
+      ;;
+    Darwin)
+      os=darwin
+      ;;
+    Linux)
+      os=linux
+      arch=$(uname -m)
+      case $arch in
+        x86_64)
+          arch=amd64
+          ;;
+        armv8*)
+          arch=arm64
+          ;;
+        aarch64*)
+          arch=arm64
+          ;;
+        armv*)
+          arch=arm
+          ;;
+        amd64|arm64)
+          arch=$arch
+          ;;
+        *)
+          echo "unknown architecture: $arch" >&2
```

Comment:
I know I worded this differently before, but maybe "unsupported" is better here? We know the architecture, it's just not supported.

```suggestion
          echo "unsupported architecture: $arch" >&2
```